app/controllers/leaderboards_controller.rb

```diff
@@ -0,0 +1,18 @@
+class LeaderboardsController < ApplicationController
+  before_action :set_quiz
+
+  def index
+    @entries = @quiz.leaderboard_entries
+                    .select("name, MAX(score) AS score")
+                    .group(:name)
+                    .order("score DESC")
+                    .limit(10)
+  end
+
+private
+
+  def set_quiz
+    quiz_id = params[:quiz_id] || params[:id]
+    @quiz = Quiz.find(quiz_id)
```

Comment:
can you just do `@quiz = Quiz.find(params[:id])` since there's only one endpoint and you're using `on: :member` in the definition

Reply:
If you take the routes suggestion 

```
resources :leaderboard_entries, only: %i[index create], on: :member

```
this will always be `params[:quiz_id]`

you can see your routes if you run `rails routes` the relevant ones here would be 

```
quiz_leaderboard_entries GET    /quizzes/:quiz_id/leaderboard_entries(.:format)                                        leaderboard_entries#index {on: :member}
POST   /quizzes/:quiz_id/leaderboard_entries(.:format)                                                 
 leaderboard_entries#create {on: :member}
```